src/api/Metadata.js

```diff
@@ -501,23 +519,46 @@ class Metadata extends File {
         }
 
         try {
+            // Metadata instances
```

Comment:
This method is turning into a bit of a monster. Can it be broken out into multiple methods?